packages/@aws-cdk/aws-kinesisfirehose-alpha/lib/source.ts

```diff
@@ -0,0 +1,79 @@
+import { Construct } from 'constructs';
+import { CfnDeliveryStream } from 'aws-cdk-lib/aws-kinesisfirehose';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as kinesis from 'aws-cdk-lib/aws-kinesis';
+
+/**
+ * A Kinesis Data Firehose delivery stream source configuration.
+ *
+ * @internal
+ */
+export interface SourceConfig {
```

Comment:
I'm not quite sure what this interface is for, but that may be that I'm not entirely familiar with Kinesis. It seems like it'll act as a way to store multiple source configurations, and append only specified parameters when binded. Can both of these attributes (and more if other sources are eventually supported) be set at the same time for a delivery stream? We want to be sure this is the right pattern to use for this implementation.

Reply:
> It seems like it'll act as a way to store multiple source configurations

Yes, this is a similar pattern used for `destinations` where for each source, the property configuration contains differing properties so instead of returning a union type of each of the different possible configurations, we return a `SourceConfig` which will only have one of these optional property configs set. 

The appropriate property config is set inside of the `_bind` methods which will be specific to each source type. This ensures that the other property configs will not be set, and it's injected directly into the L1 when it's created like so:

```ts
    const destinationConfig = props.destination.bind(this, {});
    const sourceConfig = props.source?._bind(this, this._role?.roleArn);

    const resource = new CfnDeliveryStream(this, 'Resource', {
      deliveryStreamEncryptionConfigurationInput: encryptionConfig,
      deliveryStreamName: props.deliveryStreamName,
      deliveryStreamType: props.source ? 'KinesisStreamAsSource' : 'DirectPut',
      ...sourceConfig,
      ...destinationConfig,
    });
```

If the properties in the property configs for each source were the same, then this would not be needed but that's unfortunately not the case here. It's not the most ideal or pretty solution but it's what works internally so that users won't have to think about it.